dali/operators/generic/slice/slice_attr.h

```diff
@@ -33,28 +33,36 @@
 
 namespace dali {
 
-class SliceAttr {
+class NamedSliceAttr {
  public:
-  explicit inline SliceAttr(const OpSpec &spec)
+  explicit inline NamedSliceAttr(const OpSpec &spec,
+                                 const char* kStartName = "start",
+                                 const char* kRelStartName = "rel_start",
+                                 const char* kEndName = "end",
+                                 const char* kRelEndName = "rel_end",
+                                 const char* kShapeName = "shape",
+                                 const char* kRelShapeName = "rel_shape",
+                                 const char* kAxesName = "axes",
+                                 const char* kAxisNamesName = "axis_names")
```

Comment:
These parameters are not constant expressions, as the name suggests and the logic of this function doesn't require them to outlive the call.
```suggestion
                                 const char* start_name = "start",
                                 const char* rel_start_name = "rel_start",
                                 const char* end_name = "end",
                                 const char* rel_end_name = "rel_end",
                                 const char* shape_name = "shape",
                                 const char* rel_shape_name = "rel_shape",
                                 const char* axes_name = "axes",
                                 const char* axis_names_name = "axis_names")
```